autoendpoint/src/routers/fcm/router.rs

```diff
@@ -175,24 +175,30 @@ impl Router for FcmRouter {
             .get(&app_id)
             .ok_or_else(|| FcmError::InvalidAppId(app_id.clone()))?;
 
-        // GCM is the older message format for android, and it's not possible to generate
-        // new test keys. As of 2023-09, GCM should use FCM OAuth2 in order to authenticate.
-        // FCM operates by using the more complex token as part of an OAuth2
-        // transaction. According to the documentation, GCM and FCM are interoperable,
-        // meaning that the client provided tokens should match up, and that the structure
-        // of the data should also not make a huge difference.
-        //
-        // In Oct of 2023, "legacy" GCM support was deprecated. After some
-        // research, it was hypothesised that the current FCM project may be able
-        // to accept GCM routing identifiers as FCM message tokens. The current code presumes
-        // that this is a fragile relationship and so GCM errors do not immediately drop the
-        // remote user's endpoint (potentially causing a UA reset.)
-        //
-        let platform = if client.is_gcm {
-            "gcm_as_fcmv1"
-        } else {
-            "fcmv1"
-        };
+        if client.is_gcm {
+            // GCM is the older message format for android, and it's not possible to generate
+            // new test keys.
+            // As of 2023-09-22 Legacy GCM messages are no longer supported.
+            // Due to a very unfortunate set of decisions made at the beginning of the WebPush
+            // project, we do not have access to the original GCM credential set and therefore
+            // can not move these GCM users to FCM. We have no choice but to treat this as
+            // a bridge server rejection, disable the subscription, and hope that
+            // the remote UA is running Fenix+, which will perform a daily check-in
+            // note the discrepancy and recreate the subscriptions, this time as
+            // proper FCM.
+            return Err(handle_error(
+                RouterError::NotFound,
+                &self.metrics,
+                self.db.as_ref(),
+                "gcm_as_fcmv1",
```

Comment:
nit: we're abandoning yesterday's attempt anyway
```suggestion
                "gcm",
```